File: API/Features/Authentication/Endpoints/AdminLogin.cs
namespace API.Features.Authentication.Endpoints;

public static class AdminLogin
{
    public static async Task<IResult> HandleAsync(AuthService authService, string username, string passwordHash)
    {
        if (await authService.AdminCredentialsAreValid(username, passwordHash))
        {
            string token = authService.GenerateAdminToken(username);
            return Results.Ok(new { Token = token, Role = "admin" });
        }
        return Results.Unauthorized();
    }
}

File: Web/src/lib/client/Authentication/AdminLogin.ts
import { createEndpoint } from "../api";
export type AdminLoginRequest = {
    readonly username: string;
    readonly passwordHash: string;
};
export type AdminLoginResponse = {
    readonly token: string;
};
export const AdminLogin = createEndpoint<AdminLoginRequest, AdminLoginResponse>('/api/admin/login', 'POST');
